src/pages/index.js

```diff
@@ -40,89 +50,222 @@ const enableValidation = (config) => {
+    .then((res) => {
+      return res.map((card) => ({
+        name: card.name,
+        link: card.link,
+        _id: card._id,
+      }));
+    })
+    .catch((err) => {
+      console.error(err);
+      return [];
+    });
 }
+
+cardListData()
+  .then((data) => {
+    console.log("data", data);
+    const cardElement = data.map((cardData) => createCard(cardData));
+    cardList.renderItems(cardElement);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+/* ------------------------------ cardAddPopup ------------------------------ */
+
+const cardAddPopup = new PopupWithForm({
+  popupSelector: "#card-add-modal",
+  handleFormSubmit: handleCardAddSubmit,
+});
+cardAddPopup.setEventListeners();
+
+cardAddButton.addEventListener("click", () => {
+  cardAddPopup.open();
+});
+
 function handleCardAddSubmit(formValues) {
   const name = formValues.title;
   const link = formValues.link;
-  renderCard({ name, link });
+  cardAddPopup.renderLoading(true);
+  api
+    .addNewCard({ name, link })
+
+    .then((card) => {
+      const cardElement = createCard(card);
+      cardList.addItem(cardElement);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      cardAddPopup.renderLoading(false);
+    });
+
   cardAddPopup.close();
   cardAddForm.reset();
   formValidators[cardAddForm.getAttribute("id")].disableButton();
 }
-/* -------------------------------------------------------------------------- */
-/*                               Event Listeners                              */
-/* -------------------------------------------------------------------------- */
-/* ------------------------------ Profile Edit ------------------------------ */
-profileEditButton.addEventListener("click", () => {
-  fillProfileForm();
-  //editFormValidator.resetValidation();
-  formValidators[profileEditForm.getAttribute("id")].resetValidation();
-  profileEditPopup.open();
-});
-/* ---------------------------------- Card ---------------------------------- */
-cardAddButton.addEventListener("click", () => {
-  //cardAddFormValidator.resetValidation();
-  cardAddPopup.open();
+/* ----------------------------- cardImagePopup----------------------------- */
+const cardImagePopup = new PopupWithImage("#image-modal");
+cardImagePopup.setEventListeners();
+
+function handleImageClick(name, link) {
+  cardImagePopup.open(name, link);
+}
+
+/* --------------------------- DeleteConfirmPopup --------------------------- */
+const deleteConfirmPopup = new PopupWithConfirmation({
+  popupSelector: "#confirmation-modal",
+  handleFormSubmit: handleDeleteClick,
 });
+deleteConfirmPopup.setEventListeners();
+
+function handleDeleteClick(card) {
+  deleteConfirmPopup.open();
+  deleteConfirmPopup.confirmDelete(() => {
+    api
+      .deleteCard(card.getId())
+      .then(() => {
+        card.removeCard();
+        deleteConfirmPopup.close();
+      })
+      .catch((err) => {
+        console.error("Card deletion API failed:", err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors